src/Grok.Net/Grok.cs

```diff
@@ -20,11 +20,18 @@ public class Grok
         private static readonly Regex _grokRegexWithType = new Regex("%{(\\w+):(\\w+):(\\w+)?}", RegexOptions.Compiled);
         private static readonly Regex _grokWithoutName = new Regex("%{(\\w+)}", RegexOptions.Compiled);
 
+        public IList<(string, string)> _patternViolations;
+        public IList<(string, string)> PatternViolations => _patternViolations;
+        
         public Grok(string grokPattern)
         {
             _grokPattern = grokPattern;
             _patterns = new Dictionary<string, string>();
             _typeMaps = new Dictionary<string, string>();
+            _patternViolations = new List<(string, string)>();
+            
+            // loads custom patterns from assembly reference in default constructor...
```

Comment:
Yeah I saw this. It was a bit misleading having FILE access and Reflection for loading default patterns in the default constructor. I thought this might slow the whole thing down, because it:

- uses reflection to locate manifests and iterates them
- loads at least one file from harddisk
- parses every line of this file and validates it

and there is no way to prevent this, whereas it could also be hardcoded / "compile time":

```c#
        public Grok(string grokPattern)
        {
            _grokPattern = grokPattern;
            _patterns = new Dictionary<string, string>
            {
                {"USERNAME", "[a-zA-Z0-9._-]+"},
                {"USER", "%{USERNAME}"},
                {"EMAILLOCALPART", "[a-zA-Z][a-zA-Z0-9_.+-=:]+"}
                // ...
            };
            _typeMaps = new Dictionary<string, string>();
            
            // LoadPatterns();
        }
```

But this is nothing that worries me... it felt just a bit overengineered.